File: src/main/java/compiler/block/structures/objects/MethodCallBlock.java
package compiler.block.structures.objects;

import compiler.Parameter;
import compiler.block.Block;

/**
 * Calling of a method within the class
 */
public class MethodCallBlock extends Block {

    private String name, type;
    private Parameter[] params;

    public MethodCallBlock(Block superBlock, String name, String type, Parameter[] params) {
        super(superBlock, false, false);

        this.name = name;
        this.type = type;
        this.params = params;

    }

    public String getName() {
        return name;
    }

    @Override
    public String getValue() {
        return null;
    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return "";
    }

    @Override
    public String getClosingCode() {
        return "";
    }

    @Override
    public String getBodyCode() {
        return name+"();";
    }

    public Parameter[] getParameters() {
        return params;
    }

    @Override
    public void run() {

    }

}

File: src/main/scala/compiler/block/structures/objects/MethodCallBlock.scala
package compiler.block.structures.objects

import compiler.Parameter
import compiler.block.Block

/**
  * Calling of a method within the class
  */
class MethodCallBlock(val superBlock: Block, var nameInit: String, var typeInit: String, var paramsInit: Array[Parameter]) extends Block(superBlock, false, false) {

  val name = nameInit
  val `type` = typeInit
  val params = paramsInit

  def getParameters: Array[Parameter] = {
    return params
  }

  def init() {
  }

  def getName: String = {
    return name
  }

  def getValue: String = {
    return null
  }

  def getType: String = {
    return `type`
  }

  def getOpeningCode: String = {
    return ""
  }

  def getBodyCode: String = {
    return name + "();"
  }

  def getClosingCode: String = {
    return ""
  }

  override def toString: String = {
    var paramString: String = ""
    for (parameter <- params) {
      paramString += parameter.getType + ":" + parameter.getName + "; "
    }
    return "method call: " + name + " ( " + paramString + ")"
  }
}